Deprecate LoggingTaskListener

DaveCTurner opened the pull request:
As noted in #104278 this logging is fairly unusable since it only
mentions the task ID. It dates back to the early days of the reindex
feature and was introduced before the task management APIs, but the task
management APIs are today the preferred mechanism for monitoring
long-running tasks. This commit deprecates the logger and introduces a
setting that allows it to be disabled, in preparation for its removal in
the next major version.

Closes #104278

elasticsearchmachine commented:
Hi @DaveCTurner, I've created a changelog YAML for you. *Note* that since this PR is labelled `>deprecation`, you need to update the changelog YAML to fill out the extended information sections.

elasticsearchmachine commented:
Pinging @elastic/es-distributed (Team:Distributed)

DaveCTurner committed:
Deprecate LoggingTaskListener

As noted in #104278 this logging is fairly unusable since it only
mentions the task ID. It dates back to the early days of the reindex
feature and was introduced before the task management APIs, but the task
management APIs are today the preferred mechanism for monitoring
long-running tasks. This commit deprecates the logger and introduces a
setting that allows it to be disabled, in preparation for its removal in
the next major version.

Closes #104278
